### ISSUE TITLE
Error when registering custom class using cloudpickle 0.5.1.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#1199]
The script below fails with cloudpickle==0.5.1.

```python
import ray

ray.init()

class Foo(object):
    pass

ray.put(Foo())
```

It prints the following in the background.

```
Traceback (most recent call last):
  File "/Users/rkn/Workspace/ray/python/ray/worker.py", line 1554, in fetch_and_execute_function_to_run
    function = pickle.loads(serialized_function)
AttributeError: Can't get attribute 'Foo' on <module '__main__' from '/Users/rkn/Workspace/ray/python/ray/workers/default_worker.py'>
```

**Comment by robertnishihara** [comment]
See cloudpipe/cloudpickle#131

**Comment by robertnishihara** [comment]
This was causing a test failure, which is fixed by #1200.

Of course, the error still happens with 0.5.1, but that isn't fixable.

**Comment by aldanor** [comment]
This has been apparently fixed in cloudpickle 0.5.2, could it be updated?

**Comment by robertnishihara** [comment]
@aldanor you mean change our `setup.py` to require `cloudpickle==0.5.2`? That seems like a good idea.
